### ISSUE TITLE
ae_test1

### ISSUE THREAD

**Issue by just-sparta** [just-sparta/AE#1]
import numpy as np
import pandas as pd
import seaborn as sns

from keras.datasets import mnist
from keras.models import Model
from keras.layers import Dense, Input, Reshape, Flatten
import matplotlib.pyplot as plt

(X_train, y_train), (X_test, y_test) = mnist.load_data()

X_train = X_train.astype('float32') / 255.
X_test  = X_test .astype('float32') / 255.
X_train = np.reshape(X_train, (len(X_train), 28, 28, 1))
X_test  = np.reshape(X_test,  (len(X_test),  28, 28, 1))

def deep_ae_model():

    
    input_img = Input(shape=(28, 28, 1))
    flat_img = Flatten()(input_img)
    x = Dense(128, activation='relu')(flat_img)
    encoded = Dense(2, activation='linear')(x)

    
    input_encoded = Input(shape=(2,))
    x = Dense(128, activation='relu')(input_encoded)
    flat_decoded = Dense(28 * 28, activation='sigmoid')(x)
    decoded = Reshape((28, 28, 1))(flat_decoded)

    
    encoder = Model(input_img, encoded, name="encoder")
    decoder = Model(input_encoded, decoded, name="decoder")
    autoencoder = Model(input_img, decoder(encoder(input_img)), name="autoencoder")
    return encoder, decoder, autoencoder

d_encoder, d_decoder, d_autoencoder = deep_ae_model()
d_autoencoder.compile(optimizer='adadelta', loss='binary_crossentropy')

#%%

# tensorboard --logdir=/tmp/autoencoder

#from keras.callbacks import TensorBoard

d_autoencoder.fit(X_train, X_train,
                epochs=50,
                batch_size=128,
                verbose=2,
                shuffle=True)
                #callbacks=[TensorBoard(log_dir='/tmp/autoencoder')])

#%%

encoded_imgs = d_encoder.predict(X_test)
decoded_imgs = d_decoder.predict(encoded_imgs)

n = 10  
plt.figure(figsize=(20, 4))
for i in range(n):
    
    ax = plt.subplot(2, n, i + 1)
    plt.imshow(X_test[i].reshape(28, 28))
    plt.gray()
    ax.get_xaxis().set_visible(False)
    ax.get_yaxis().set_visible(False)

   
    ax = plt.subplot(2, n, i + 1 + n)
    plt.imshow(decoded_imgs[i].reshape(28, 28))
    plt.gray()
    ax.get_xaxis().set_visible(False)
    ax.get_yaxis().set_visible(False)
plt.show()

#%%

def show(img):
    plt.imshow(img, cmap = "gray", interpolation = "none")

#ОТОБРАЖЕНИЕ MANIFOLD
X_proj = d_encoder.predict(X_train[:10000])
X_proj.shape

proj = pd.DataFrame(X_proj)
proj.columns = ["y", "x"]
proj["labels"] = y_test[:10000]
sns.lmplot("y", "x",hue = "labels", data = proj, fit_reg=False)   
    
#%%

#ДВИГАЕМСЯ ПО MANIFOLD ПО Х:
for i in range(64):
    plt.subplot(8,8,i+1)
    pt = np.array([[i/3,0]])
    show(d_decoder.predict(pt).reshape((28, 28)))
    plt.xticks([])
    plt.yticks([])
